examples/api/lib/widgets/inherited_model/inherited_model.0.dart

```diff
@@ -0,0 +1,169 @@
+    required super.child,
+  });
+
+  final double fooSize;
+  final double barSize;
+
+  static FooBarModel? of(BuildContext context, String aspect) {
+    return InheritedModel.inheritFrom<FooBarModel>(context, aspect: aspect);
+  }
+
+  @override
+  bool updateShouldNotify(FooBarModel oldWidget) {
+    return fooSize != oldWidget.fooSize
+     || barSize != oldWidget.barSize;
+  }
+
+  @override
+  bool updateShouldNotifyDependent(FooBarModel oldWidget, Set<String> dependencies) {
+    if (fooSize != oldWidget.fooSize && dependencies.contains('foo')) {
+      print('Rebuilding Foo Widget');
+      return  true;
+    }
+    if (barSize != oldWidget.barSize && dependencies.contains('bar')) {
+      print('Rebuilding Bar Widget');
+      return  true;
+    }
+    return false;
+  }
+}
+
+class InheritedModelExample extends StatefulWidget {
+  const InheritedModelExample({super.key});
+
+  @override
+  State<InheritedModelExample> createState() => _InheritedModelExampleState();
+}
+
+class _InheritedModelExampleState extends State<InheritedModelExample> {
+  double _fooSize = 50.0;
+  double _barSize = 50.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('InheritedModel Sample')),
+      body: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              FooBarModel(
+                fooSize: _fooSize,
+                barSize: _barSize,
+                child: const FooWidget(),
+              ),
+              const SizedBox(height: 10.0),
+              ElevatedButton(
+                onPressed: () {
+                  setState(() {
+                    _fooSize = _fooSize == 50.0 ? 100.0 : 50.0;
+                  });
+                },
+                child: const Text('Resize Foo'),
+              ),
+
+            ],
+          ),
+          const SizedBox(width: 20.0),
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              FooBarModel(
+                fooSize: _fooSize,
+                barSize: _barSize,
+                child: const BarWidget(),
+              ),
+              const SizedBox(height: 10.0),
+              ElevatedButton(
+                onPressed: () {
+                  setState(() {
+                    _barSize = _barSize == 50.0 ? 100.0 : 50.0;
+                  });
+                },
+                child: const Text('Resize Bar'),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class FooWidget extends StatelessWidget {
+  const FooWidget({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final FooBarModel? fooBarModel = FooBarModel.of(context, 'foo');
+    final Color randomColor = Colors.primaries[Random().nextInt(Colors.primaries.length)];
```

Comment:
Using ransom here is strange and makes the app indeterministic. It's not good practices and we shouldn't show that in our examples which should demonstrate good practices.